### PR TITLE
fix: Fix CalendarView going into an infinite loop

### LINKED ISSUE #11830: [WASM] CalendarView.SelectedDates not working
### Current behavior

Please see attached repo (I also left some comments in sources): the selected date would not been seen/selected on WASM. 

[CalendarView.zip](https://github.com/unoplatform/uno/files/11092673/CalendarView.zip)

### Expected behavior

WASM should work as WindowsAppSDK

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #11830

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewGeneratorHost.cs [hunk 1]
```diff
@@ -350,7 +350,7 @@ internal DateTime GetDateAt(uint index)
 		internal int CalculateOffsetFromMinDate(DateTime date)
 		{
 			var pIndex = 0;
-			DateTime estimatedDate = m_lastVisitedDateAndIndex.first.ToUniversalTime();
+			DateTime estimatedDate = m_lastVisitedDateAndIndex.first; //.ToUniversalTime(); - UNO
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewGeneratorHost.cs:353]: @dr1rrb is there an abstraction in the runtime tests for this control to help timezone changes?

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewGeneratorHost.cs:353]: Don't think so

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewGeneratorHost.cs:353]: Looking at the code, this is probably the way to go : https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.calendarview.calendaridentifier?view=winrt-22621

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewGeneratorHost.cs:353]: @dr1rrb Could we merge without a test?

Or, could we do [this](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/change-time-zone?view=azure-devops&tabs=preview-page)?

**Comment by TopProgrammer77** [comment]
https://github.com/unoplatform/uno/pull/12179/checks?check_run_id=13272987787

It might look like the image below.

image

However, this problem has been existing from before.

**Comment by Youssef1313** [comment]
@TopProgrammer77 I'm not following with this.

**Comment by TopProgrammer77** [comment]
OK, I will check this in updated version

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**